pkg/storage/snapshot/snapshot.go

```diff
@@ -218,10 +229,18 @@ func (ctrl *VMSnapshotController) updateVMSnapshot(vmSnapshot *snapshotv1.Virtua
 		}
 	}
 
-	if err = ctrl.updateSnapshotStatus(vmSnapshot, content, source); err != nil {
+	vmSnapshot, err = ctrl.updateSnapshotStatus(vmSnapshot)
+	if err != nil {
 		return 0, err
 	}
 
+	if vmSnapshotDeleting(vmSnapshot) && canRemoveFinalizer {
```

Comment:
Hi,
At this point of the flow, `(!vmSnapshotProgressing(vmSnapshotCpy) || contentDeletedIfNeeded(vmSnapshotCpy, content))` is always evaluated to true? I am asking because of the previous condition in update status:
```
	if vmSnapshotDeleting(vmSnapshotCpy) {
		// Enable the vmsnapshot to be deleted only in case the source is unlocked AND
		// (the operation completed completed OR
		// after waiting until the content is deleted if needed)
		if canRemoveFinalizer && (!vmSnapshotProgressing(vmSnapshotCpy) || contentDeletedIfNeeded(vmSnapshotCpy, content)) {
			controller.RemoveFinalizer(vmSnapshotCpy, vmSnapshotFinalizer)
		}
```

Reply:
Hi @enp0s3 thanks for the review, at this point, the snapshot `status.phase` should be `Deleting` but more importantly, the content will have already been deleted  if needed a couple lines earlier in the function here: https://github.com/mhenriks/kubevirt/blob/snapshot-api-status-subresource/pkg/storage/snapshot/snapshot.go#L223